### PR TITLE
Define __str__ for nicer plot titles for variables

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Avoid confusing class name in plot titles.

Before:
image

After:
image

### CODE REVIEW

File: python/src/scipp/__init__.py [hunk 1]
```diff
@@ -77,6 +77,18 @@ def custom_formatwarning(msg, *args, **kwargs):
 setattr(DataArray, 'sizes', property(_make_sizes))
 setattr(Dataset, 'sizes', property(_make_sizes))
 
+
+def _str_factory(cls):
+    def _str(obj):
+        return f'scipp.{cls.__name__}(sizes={obj.sizes})'
```
Comment (jl-wynen) [inline python/src/scipp/__init__.py:83]: 'dim' instead of 'sizes'?

Also, keep it consistent with the HTML repr, don't mix braces and parens.

Comment (SimonHeybrock) [reply python/src/scipp/__init__.py:83]: On the Python side, `dims` refers only to the keys (dim labels).

Is it worth the extra code to replace the `{}` that Python uses for formatting dicts?

**Review by jl-wynen — APPROVED** [review]
Looks good.

A minor point: Is is possible that some internal type like `VariableView`, aka `bucket<Variable>` gets passed to a plot?

**Comment by jl-wynen** [comment]
I think you are right.

**Pull request merged**